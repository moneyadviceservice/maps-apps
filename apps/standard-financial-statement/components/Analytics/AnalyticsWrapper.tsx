import { PropsWithChildren } from 'react';

import { Analytics } from '@maps-react/core/components/Analytics';

export type StandardFinancialStatementStep = number | string;

type Props = {
  analyticsData: Record<string, unknown>;
};

export const AnalyticsWrapper = ({
  analyticsData,
  children,
}: Props & PropsWithChildren) => {
  return (
    <Analytics
      analyticsData={analyticsData}
      currentStep={0}
      formData={{}}
      lastStep={0}
      errors={[]}
      trackDefaults={{
        toolCompletion: false,
        toolStartRestart: false,
        errorMessage: false,
        pageLoad: true,
        emptyToolCompletion: false,
      }}
    >
      {children}
    </Analytics>
  );
};
