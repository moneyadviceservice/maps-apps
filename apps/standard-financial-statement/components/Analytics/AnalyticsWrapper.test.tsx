import React from 'react';

import { render } from '@testing-library/react';

import { AnalyticsWrapper } from './AnalyticsWrapper';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
  }),
}));

describe('AnalyticsWrapper', () => {
  it('renders children inside Analytics', () => {
    const { container } = render(
      <AnalyticsWrapper analyticsData={{ page: { pageName: 'Test Page' } }}>
        <div>Test Child</div>
      </AnalyticsWrapper>,
    );
    expect(container).toMatchSnapshot();
  });
});
