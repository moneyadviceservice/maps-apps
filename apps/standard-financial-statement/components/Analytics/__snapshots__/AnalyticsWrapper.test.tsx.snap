// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalyticsWrapper renders children inside Analytics 1`] = `
<div>
  <div>
    Test Child
  </div>
</div>
`;
